.github/workflows/ci.yml

```diff
@@ -3,44 +3,70 @@ name: CI
 on: pull_request
 
 jobs:
-  tests:
-
-    name: Test Python ${{ matrix.python_version }}
+  lint:
+    name: Lint
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python_version: ['3.6', '3.7', '3.8', '3.9']
-    env:
-      PYTHONDEVMODE: 1
 
     steps:
     - uses: actions/checkout@v2
 
     - name: Set up Python
       uses: actions/setup-python@v2
       with:
-        python-version: ${{ matrix.python_version }}
+        python-version: '3.x'
+        cache: 'pip'
+        cache-dependency-path: 'pyproject.toml'
 
-    - name: Poetry setup
+    - name: Install Flit
       run: |
-        python -m pip install poetry
+        python -m pip install flit
 
     - name: Install dependencies
       run: |
-        poetry install
+        flit install --extras dev
 
     - name: Lint with flake8
       run: |
-        poetry run flake8
+        flake8
 
     - name: Lint with mypy
       run: |
-        poetry run mypy mousebender
+        mypy mousebender
 
     - name: Check format with Black
       run: |
-        poetry run black --check .
+        black --check .
+
+    - name: Can be built
+      run: |
+        flit build
+
+  tests:
+
+    name: Test Python ${{ matrix.python_version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python_version: ['3.6', '3.7', '3.8', '3.9', '3.10']
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python_version }}
+        cache: 'pip'
+        cache-dependency-path: 'pyproject.toml'
+
+    - name: Install Flit
+      run: |
+        python -m pip install flit
+
+    - name: Install dependencies
+      run: |
+        flit install --extras test
```

Comment:
It seemed to be on my local machine that `flit install` installed everything - was that not the case? (When I tried `flit install --extras test` all was already there?)